besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2133,7 +2134,10 @@ private void issueOptionWarnings() {
             "--remote-connections-max-percentage"));
 
     // Check that block producer options work
-    if (!isMergeEnabled()) {
+    if (!isMergeEnabled()
+        && !(Optional.ofNullable(genesisConfigOptions)
+            .orElseGet(() -> new StubGenesisConfigOptions())
+            .isPoa())) {
```

Comment:
Wonder if we can get away with `!isMergeEnabled && isEthHash` instead?

Note, currently public networks genesis' are either clique or ethash, but future public testnets (proof of stake at genesis) might have the consensus unspecified. These networks shouldn't care about miner-enabled either I think.

Reply:
Had a play and I think that `!isMergeEnabled && isEthHash` is what we want for both checks, i.e. the `miner-stratum-enabled` one underneath this (I think it's a bug it doesn't have `!isMergeEnabled`)

PoS at genesis networks that omit the consensus engine will default to "unknown" in the besu code, so it works for them too.

Maybe eventually it can simply be `isEthHash` but might have to always support a "merged" mainnet config (where it was originally ethash).

Reply:
...so yes, the isPoa is probably redundant for this, though potentially useful for other things at some point?

Reply:
Thanks @siladu yes for this case I think `isEthHash()` makes sense. Maybe it's worth leaving `isPoa()` as a general utility that could help make future code changes/logic a little clearer. I'll submit a new commit that changes the current logic but leaves `isPoa()` in. Let me know what you think.

Reply:
Realised that `getActualGenesisConfigOptions()` provides the combined output of a genesis file, network config, + overrides, so no refactoring required.

Reply:
@siladu I think I'm happy with the current shape of the PR:

- switched to using `isEthash()`
- moved `--miner-stratum` option warning check inside same block as `--miner-enabled`
- left `isPoa()` in as a helper function
- added some new unit tests for QBFT & IBFT2 option checking

Ready for you to take another look at when you have a mo.